includes/Core/Util/Entity_Factory.php

```diff
@@ -60,8 +60,15 @@ public static function from_context() {
 
 		// Otherwise, run frontend-specific `WP_Query` logic.
 		if ( $wp_the_query instanceof WP_Query ) {
-			return self::from_wp_query( $wp_the_query );
+			$entity = self::from_wp_query( $wp_the_query );
+
+			if ( is_null( $entity ) || ! defined( 'AMP__VERSION' ) ) {
+				return $entity;
+			}
+
+			return self::maybe_convert_to_amp_entity($_SERVER['REQUEST_URI'], $entity); // phpcs:ignore
```

Comment:
@kuasha420, please, use the same approach to get the `REQUEST_URI` value as we do it in here: https://github.com/google/site-kit-wp/blob/2f43c80560510e50bd5a3af914baa52f48f143c2/includes/Modules/Site_Verification.php#L98

It will eliminate a need to use `// phpcs:ignore`. To get the context instance, you can use `Plugin::instance()->context()->...`.